### PR TITLE
Fix spurious .sock files running envelope unit tests

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

Removes sock files left by the fake grpc kms provider

xref https://github.com/kubernetes/kubernetes/issues/78982

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
cc @dims @immutableT 

/priority backlog
/sig auth

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 1]
```diff
@@ -410,6 +413,16 @@ func startFakeKMSProvider(version, endpoint string) (*fakeKMSPlugin, error) {
...
 	apiVersion string
 	server     *grpc.Server
+	sockFile   string
+}
+
+func (s *fakeKMSPlugin) Stop() {
+	// Stop the server
+	s.server.Stop()
+	// If this isn't a Linux abstract namespace socket, or if we're on a non-linux platform, clean up the socket file
+	if !strings.HasPrefix(s.sockFile, "@") || runtime.GOOS != "linux" {
```
Comment (dims) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go]: @liggitt the file is marked `// +build !windows` ... do we leave it as-is?

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go]: yes

**Comment by spiffxp** [comment]
/lgtm
/retest
test failure appears unrelated

**Pull request merged**